sklearn/utils/themes.py

```diff
@@ -0,0 +1,237 @@
+"""
+The :mod:`sklearn.utils.themes` module provides a
+CSS template class for creating dynamic stylesheets.
+It exposes utilities and themes for styling displayed diagrams.
+"""
+from string import Template
+
+
+class CssTemplate(Template):
+    """Create parameterized CSS templates with `$$` as the variable prefix."""
+
+    delimiter = "$$"
+
+
+_AUTO_TEMPLATE = """
+$$light
+
+@media (prefers-color-scheme: dark) {
+    $$dark
+}
```

Comment:
Currently, the `light` and `dark` variables contains all the color and positional information. Given that the colors need to change, I think it's better to use [CSS variables](https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties) for the colors. Concretely, for the auto option:

```css
:root {
  --sklearn-color-1: black;
}
@media (prefers-color-scheme: dark) {
  :root {
    --sklearn-color-1: #788aa6;
  }
}

#$id {
  color: var(--sklearn-color-1);
}
```

### Future (not for this PR)

In the future, the CSS variables also makes it easier to give user control over the theme. For example, if the user request the dark theme:

```css
:root {
 --sklearn-color-1: #788aa6;
}

/* everything below this line is the same as the above example */
#$id {
  color: var(--sklearn-color-1);
}
```

In Python, we can just add two string together to get new css styles because only the "color header" changes.